src/DataProvider/ChainCollectionDataProvider.php

```diff
@@ -39,8 +39,14 @@ public function getCollection(string $resourceClass, string $operationName = nul
     {
         foreach ($this->dataProviders as $dataProvider) {
             try {
+                if ($dataProvider instanceof RestrictedDataProviderInterface
+                    && !$dataProvider->supports($resourceClass, $operationName)) {
+                    continue;
+                }
+
                 return $dataProvider->getCollection($resourceClass, $operationName);
             } catch (ResourceClassNotSupportedException $e) {
+                @trigger_error(sprintf('Throwing a "%s" in a data provider is deprecated in favor of implementing "%s"', ResourceClassNotSupportedException::class, RestrictedDataProviderInterface::class), E_USER_DEPRECATED);
```

Comment:
@soyuka I misunderstood your comment at first. I hope you all see how easier it would have been had each argument been on its own line: you could have commented directly under `get_class($e)`. BTW, the sf standard is to have the message on one line, but is the `sprintf` part of the message? If yes, that's really a stupid rule who just made us loose precious time. It just makes life harder for everyone for no good reason.